lis/surfacemodels/land/noahmp.4.0.1/irrigation/noahmp401_getirrigationstates.F90

```diff
@@ -175,341 +178,645 @@ subroutine noahmp401_getirrigationstates(n,irrigState)
      otimee = otimefs + irrhrf
   endif
   
-
-
-  do i=1,LIS_rc%npatch(n,LIS_rc%lsm_index)/LIS_rc%nensem(n)
-
-     sfctemp_avg = 0.
-     shdfac_avg  = 0.
-     smc_avg     = 0.
-
-     do m=1,LIS_rc%nensem(n)
-
-        t=(i-1)*LIS_rc%nensem(n)+m
-        
-        sfctemp_avg=sfctemp_avg+NOAHMP401_struc(n)%noahmp401(t)%sfctmp        
-        shdfac_avg=shdfac_avg+NOAHMP401_struc(n)%noahmp401(t)%fveg            
-        
-        do k=1,nsoil
-           
-           smc_avg(k)=smc_avg(k)+NOAHMP401_struc(n)%noahmp401(t)%smc(k)
-        
-        end do
+  if(LIS_rc%pert_bias_corr.eq.0) then
+  	  do i=1,LIS_rc%npatch(n,LIS_rc%lsm_index)
+		 timestep = NOAHMP401_struc(n)%dt
+		 soiltyp = noahmp401_struc(n)%noahmp401(t)%soiltype
```

Comment:
loop over i --> where is i used? How is t set when it enters at this line? (I would have expected a mapping from i to t somewhere)

Reply:
Great catch. Exactly what was missing. Louise had it in a subsequent bug fix commit in one of her branches.